src/data_api/reporting.se

```diff
@@ -27,186 +27,161 @@ extern branches: [addCurrency:[int256,int256,int256,int256]:int256, addMarketToB
 REPCONTRACT = self.controller.lookup('repContract')
 extern repContract: [allowance:[address,address]:uint256, approve:[address,uint256]:uint256, balanceOf:[address]:uint256, decimals:[]:uint256, getRidOfDustForLaunch:[]:int256, getSeeded:[]:uint256, name:[]:uint256, setSaleDistribution:[address[],uint256[]]:uint256, symbol:[]:uint256, totalSupply:[]:uint256, transfer:[address,uint256]:uint256, transferFrom:[address,address,uint256]:uint256]
 
+inset('../macros/float.sem')
 inset('../macros/refund.sem')
 
 # Storage of all data associated with reporters
-    # Reporting index is the branch
-    # Reputation index is the index of the reporter
-    # RepIDtoIndex returns a reporter's reputation[] index given their reputationID as the key
-    # total rep is all the rep in augur
-    # active rep is the fxpAmount which is actually active
-    # fork is the child branch of a parent which forked
-    # fxpDormant rep stores fxpDormant rep values
-    # reportedOnNonFinalRoundTwoEvent is the eventID if a person reported on a round 2 event before it was in the second round [i.e. the first reporting backstop], they cannot convert their rep to dormant or send rep until they've finished the resolution process for that round 2 event, set this to 0 once the event is final
-data Reporting[](reputation[](repValue, reporterID), numberReporters, repIDtoIndex[], totalRep, dormantRep[](repValue, reporterID), activeRep, reportedOnNonFinalRoundTwoEvent[])
+    # Reporting is a map of branch ID to branch details
+    # activeRep is a map of reporter address to attorep
+    # dormantRep is a map of reporter address to attorep. NOTE: address 0x0000000000000000000000000000000000000000 contains REP that needs to be redistributed
+    # totalActiveRep is all of the active attorep in augur
+    # totalDormantRep is all of the dormant attorep in augur (totalActiveRep + totalDormantRep is all REP in Augur)
+    # numberReporters is the number of addresses that hold non-zero rep
+    # reporterLocks is a map of reporter address to an array of events that is causing them to be locked and the number of events causing them to be locked.  If there are no events then the reporter is free to deactivate their rep.  If there are any locking events the reporter is unable to deactivate their REP.
```

Comment:
Being locked by one event is the same as being locked by n since you'll always be unlocked at the same time, so at least as it is now there's no need to store more